[MM-26973] Feature: Account setting for disabling GIF and emoji animation

Victor-Nyagudi opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does, as well as QA test steps (if applicable and if not already added to the Jira ticket).
-->
This PR adds a setting that lets users enable or disable GIFs and animated emojis from auto-playing.

##### QA Steps

1. Open the emoji picker, select a GIF from the GIF tab, and post it.
2. Verify the GIF is playing (autoplay is on by default), the pause button is hidden, and hovering over the GIF reveals it on desktop devices (clicking the GIF once reveals it on mobile devices and it's hidden again after 4 seconds).
3. Click Settings -> Display -> Autoplay GIFs and Animated Emojis -> Turn Off -> Save.
4. Verify a static GIF is shown with a play button.
5. Turn on autoplay and post a link to a GIF.
6. Repeat steps 3 and 4.
7. Turn on autoplay and upload a GIF attachment.
8. Repeat steps 3 and 4.
9. Turn on autoplay and post an animated emoji from the emoji picker.
10. Repeat step 3 and verify a static post emoji is shown.
11. Turn on autoplay and add an animated reaction to any post.
12. Repeat step 3 and verify a static reaction is shown.
13. Click the play button on any static GIF and verify a playing GIF is shown.
14. Click the pause button on any playing GIF and verify a static GIF is shown.
15. Start a thread from a static GIF and verify the original post is a static GIF. Vice versa for playing GIFs.

#### Ticket Link
<!--
If applicable, please include both or either of the following links:

Fixes https://github.com/mattermost/mattermost/issues/XXX
Jira https://mattermost.atlassian.net/browse/MM-XXX
-->

Fixes https://github.com/mattermost/mattermost/issues/17740
Jira https://mattermost.atlassian.net/browse/MM-26973%7Csmart-link

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.

For an easier comparison of UI changes a table (template below) can be used.

|  before  |  after  |
|----|----|
| <insert before screenshot here> | <insert after screenshot here> |

-->


https://github.com/user-attachments/assets/592b6a79-7159-4729-9502-21d33cf8ce78


#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates).
* API additions—new endpoint, new response fields, or newly accepted request parameters.
* Database changes (any).
* Schema migration changes. Use the [Schema Migration Template](https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?usp=sharing) as a starting point to capture these details as release notes. 
* Websocket additions or changes.
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating).
* New features and improvements, including behavioral changes, UI changes, and CLI changes.
* Bug fixes and fixes of previous known issues.
* Deprecation warnings, breaking changes, or compatibility notes.

If no release notes are required, write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```release-note
NONE
```
-->
```release-note
Added 'Autoplay GIFs and Emojis' setting to toggle when GIFs and animated emojis should play.
```

mattermost-build commented:
Hello @Victor-Nyagudi,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

Victor-Nyagudi commented:
## PR Checklist

I can confirm that:

- [ ] There are no new console errors or warnings after my changes.
- [x] Any CSS or styling used is supported on all browsers.
- [ ] If any copy or text is added, it is localized and approved by the copy team.
- [ ] I have tested my changes on mobile, tablet, and desktop viewport sizes.
- [x] I've added comments where necessary explaining why I made certain changes.
- [ ] I've read the [contribution checklist](https://developers.mattermost.com/contribute/more-info/getting-started/contribution-checklist/) on submitting PRs.
- [ ] If any tests are required, I've added them or updated the existing ones.
- [ ] There are no new lint errors. 
- [ ] There are no unnecessary re-renders negatively affecting performance.
- [ ] All tests pass after running `npm run test`.
- [x] If any QA steps are required, I've added them to the OP.
- [ ] I marked this PR ready for review after pulling the latest code from the `master` branch.

matthewbirtch commented:
@Victor-Nyagudi tag me when this is ready for review and I can test it out from a UX perspective

Victor-Nyagudi commented:
This PR looks like it requires a [feature flag](https://developers.mattermost.com/contribute/more-info/server/feature-flags/#when-to-use) but I'm not familiar with Go.

Any tips on where to go from here? (no pun intended)

Victor-Nyagudi commented:
Marking this as ready for review in the meantime for UX review.

matthewbirtch reviewed (changes requested):
Great work @Victor-Nyagudi. It's working very well. A couple things I'm noticing:

- custom status: if users apply a custom status with an animated emoji, that should be a status image if Settings > Display > Autoplay GIFs and Emojis
- animated GIF emojis that are paused: seem to be very low resolution, are not getting sized consistently, are adding extra space and causing alignment issues. See the comparison below

<table>
<tr>
<td>Animation turned on</td>
<td>Animated turned off</td>
<tr>
<tr>
<td><img width="387" alt="image" src="https://github.com/user-attachments/assets/991e2516-b301-44dc-abde-1c104babfe44" />
</td>
<td><img width="414" alt="image" src="https://github.com/user-attachments/assets/fe457229-7b78-449b-b5aa-4288614a2f74" /></td>
<tr>
</table>

- Same problem in the channel header if an animated emoji is used.
<table>
<tr>
<td>Animation turned on</td>
<td>Animated turned off</td>
<tr>
<tr>
<td>
<img width="258" alt="image" src="https://github.com/user-attachments/assets/42b4e1d7-6491-4ec1-9534-0875a1860ca8" />
</td>
<td>
<img width="271" alt="image" src="https://github.com/user-attachments/assets/ec1a598f-fb72-40a6-8362-d7ebdd763947" />
</td>
<tr>
</table>

Victor-Nyagudi commented:
I'll be away until the 20th, so I'll look into this when I return.

mattermost-build commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

Victor-Nyagudi commented:
I've been low on bandwidth lately. Planning on digging into these on Monday.

Victor-Nyagudi commented:
Still low on bandwidth. I'll provide an update when things open up.

Victor-Nyagudi commented:
I'm currently focusing on migrating class components to function components, so this will be on hold until I'm done with the other stuff.

Function components should make it easier to implement this feature.

Victor-Nyagudi commented:
No progress. Still working on the migrations.
